core/app/services/spree/products/prepare_nested_attributes.rb

```diff
@@ -130,7 +130,7 @@ def update_option_value_variants(option_value_params, existing_variant)
         option_value_variant_params = {}
 
         option_value_params.each_with_index do |opt, index|
-          option_type = Spree::OptionType.where(name: opt[:name].parameterize).first_or_initialize do |o|
+          option_type = Spree::OptionType.find_by(id: opt.require(:id)) || Spree::OptionType.where(name: opt[:name].parameterize).first_or_initialize do |o|
```

Comment:
Rails has a known issue when handling arrays of hashes with inconsistent keys (e.g., some with `id`, others without). When such structures are submitted via params, Rails may merge values incorrectly by position rather than by key, leading to attributes from one object being mistakenly applied to another.

By using `opt.require(:id)`, we enforce the presence of the `id` key and avoid silent data misalignment during param parsing. This ensures each nested object is processed with the correct data.

Rails issue: [rails/rails#23997](https://github.com/rails/rails/issues/23997)